lurk-verify/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+    ))]
+    pub proof: lurk::proof::nova::Proof<'a>,
+    pub reduction_count: ReductionCount,
+}
+
+// Either a pair of input and expected output or a commitment opening, so the verifier can check that
+// they match the scalar public inputs and outputs
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum Claim<F: LurkField> {
+    Text(String, String),
+    // TODO: Use syntax AST once implemented with LDON refactor
+    //Syntax(Syn, Syn),
+    ScalarPtr(ScalarPtr<F>, ScalarPtr<F>),
+    // TODO
+    //#[serde(bound(
+    //    serialize = "Opening<F>: Serialize",
+    //    deserialize = "Opening<F>: Deserialize<'de>"
+    //))]
+    // Opening(Opening<F>),
+}
+
+impl<'a> LurkProof<'a, S1> {
+    pub fn verify(&self, pp: &PublicParams) -> Result<VerificationResult, VerifyError> {
+        // Check if the stored input and output exprs are the same as the verifier expects
+        if let Some(claim) = &self.claim {
+            self.verify_claim(&claim)?;
+        }
+        let verified = self
+            .proof
+            .verify(
+                pp,
+                self.num_steps,
+                self.public_inputs.clone(),
+                &self.public_outputs,
+            )
+            .expect("error verifying");
+
+        let result = VerificationResult::new(verified);
+
+        Ok(result)
+    }
+
+    pub fn verify_claim(&self, claim: &Claim<S1>) -> Result<(), VerifyError> {
+        let proof_exprs = (
+            ScalarPtr::from_parts(self.public_inputs[0], self.public_inputs[1]),
+            ScalarPtr::from_parts(self.public_outputs[0], self.public_outputs[1]),
+        );
+        let verify_exprs = match claim {
+            Claim::Text(input, output) => {
+                (string_to_scalar_ptr(input)?, string_to_scalar_ptr(output)?)
+            }
+            //Claim::Syntax(syn) => todo!(),
+            Claim::ScalarPtr(scalar_input, scalar_output) => (*scalar_input, *scalar_output),
+        };
+        if proof_exprs != verify_exprs {
+            return Err(VerifyError::Verification(
+                "mismatched public inputs/outputs".into(),
+            ));
+        }
+        Ok(())
+    }
+
+    pub fn write_to_path<P: AsRef<Path>>(&self, path: P) {
+        let file = File::create(path).expect("failed to create file");
+        let writer = BufWriter::new(&file);
+
+        serde_json::to_writer(writer, &self).expect("failed to write file");
+    }
+
+    pub fn read_from_path<P: AsRef<Path>>(path: P) -> Result<Self, VerifyError> {
+        let file = File::open(path)?;
+        let reader = BufReader::new(file);
+        Ok(serde_json::from_reader(reader)?)
+    }
+
+    pub fn read_from_stdin() -> Result<Self, VerifyError> {
+        let reader = BufReader::new(io::stdin());
+        Ok(serde_json::from_reader(reader).expect("failed to read from stdin"))
+    }
+}
+
+// TODO: Ignoring env and cont for simplicity, are they necessary?
```

Comment:
I think the answer is 'no' because it's unclear that you have an env or cont to ignore in the first place.